### PR TITLE
SEPADirectDebitLauncher with v3 BrowserSwitchClient

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Update SEPADirectDebitLauncher to use browser switch pattern defined in https://github.com/braintree/browser-switch-android/pull/81

### Checklist

 - [x] Added a changelog entry
 - [x] Relevant test coverage

### Authors
@sarahkoop 

### PayPal and Braintree engineering requirements
 > If you're an engineer for PayPal or Braintree, complete this section. For PRs from the community, please ignore!

**When do these changes need to be released?** 
[ ] ASAP (let's discuss outside this PR)
[ ] Soon, here's our target release date: <DD/MM/YYYY>
[x] Whenever, no rush

### CODE REVIEW

File: Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java [hunk 1]
```diff
@@ -24,35 +26,37 @@
 public class SEPADirectDebitFragment extends BaseFragment {
 
     private SEPADirectDebitClient sepaDirectDebitClient;
-    private SEPADirectDebitLauncher sepaDirectDebitLauncher;
+    private final SEPADirectDebitLauncher sepaDirectDebitLauncher = new SEPADirectDebitLauncher();
+
+    private SEPADirectDebitPendingRequest.Started pendingRequest;
```
Comment (sarahkoop) [inline Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java:31]: Pending request storage to be updated when #883 is merged

**Review by MaxHastingsPP — APPROVED** [review]
lgtm

File: Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java [hunk 2]
```diff
@@ -74,24 +78,48 @@ public void launchSEPADirectDebit(View view) {
...
+                handleError(new UserCanceledException("User canceled SEPA Direct Debit"));
+            } else if (result instanceof SEPADirectDebitResult.Success) {
+                handleSEPANonce(((SEPADirectDebitResult.Success) result).getNonce());
             }
         });
     }
 
     private String generateRandomCustomerId() {
-        return UUID.randomUUID().toString().substring(0,20);
+        return UUID.randomUUID().toString().substring(0, 20);
```
Comment (hollabaq86) [inline Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java:113]: does a linter yell at this, is that why we added the space?

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitBrowserSwitchRequest.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    browserSwitchPendingRequest: BrowserSwitchPendingRequest.Started
+) {
+
+    private var browserSwitchPendingRequest: BrowserSwitchPendingRequest.Started
+
+    init {
+        this.browserSwitchPendingRequest = browserSwitchPendingRequest
+    }
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```
Comment (hollabaq86) [inline SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitBrowserSwitchRequest.kt:18]: just for my knowledge, what does this scope setting do?

**Review by hollabaq86 — APPROVED** [review]
looks good! Some general Qs for my knowledge

File: Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java (hunk 2 above)
Comment (sarahkoop) [reply Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java:113]: Yes this was a result of using Android Studio's auto reformat code tool (mostly breaks up newlines where necessary).

File: SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitBrowserSwitchRequest.kt (hunk 3 above)
Comment (sarahkoop) [reply SEPADirectDebit/src/main/java/com/braintreepayments/api/SEPADirectDebitBrowserSwitchRequest.kt:18]: This is our Kotlin equivalent of Java's package-private - so essentially saying this method is "internal" to our SDK an not covered by semantic versioning

**Pull request merged**